clientlibrary/worker/worker.go

```diff
@@ -235,7 +242,10 @@ func (w *Worker) eventLoop() {
 		err := w.syncShard()
 		if err != nil {
 			log.Errorf("Error getting Kinesis shards: %+v", err)
-			time.Sleep(time.Duration(w.kclConfig.ShardSyncIntervalMillis) * time.Millisecond)
+
+			// Add [-50%, +50%] random jitter to ShardSyncIntervalMillis in case of error.
+			shardSyncSleep := w.kclConfig.ShardSyncIntervalMillis/2 + rand.Intn(int(w.kclConfig.ShardSyncIntervalMillis))
```

Comment:
// Intn returns, as an int, a non-negative pseudo-random number in [0,n)
// from the default Source.
// It panics if n <= 0.
func Intn(n int) int { return globalRand.Intn(n) }

It should be:
shardSyncSleep := w.kclConfig.ShardSyncIntervalMillis/2 + rand.Intn(int(w.kclConfig.ShardSyncIntervalMillis/2))

Reply:
By doing interval/2 + rand(interval)/2, my understanding is that we would have the sleep ranging from [interval/2, interval]. The average sleep would be in that case of: 0.75 interval

My intention for the random jitter was that it, on average, the sleep would be the same as configured by the user (to not induce new behavior). That's why I set the random jitter to be from -50% to +50%, so that the interval is [interval/2, 1+interval/2].